src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1588,7 +1591,9 @@ private string ReadLineFromFile(string initialContent)
                     break;
                 }
 
-                if (c == '\b')
+                // If NoPrompt is true, we are in a sort of server mode where we shouldn't
+                // do anything like edit the command line - every character is part of the input.
+                if (c == '\b' && !NoPrompt)
```

Comment:
Not sure about that. IO redirection scenario could be used to marshal interactive session as well, even with NoPrompt. I.e. in something like EditorServices. Did this caused mentioned bugs or it's just for tests?